### PR TITLE
 First autocomplete option is now focused when typing

### PR DESCRIPTION
**Pull request by jacekbogdanski**
## What is the purpose of this pull request?

New feature

## Does your PR contain necessary tests?

All patches which change the editor code must include tests. You can always read more
on [PR testing](https://docs.ckeditor.com/ckeditor4/docs/#!/guide/dev_contributing_code-section-tests),
[how to set the testing environment](https://docs.ckeditor.com/ckeditor4/docs/#!/guide/dev_tests) and
[how to create tests](https://docs.ckeditor.com/ckeditor4/docs/#!/guide/dev_tests-section-creating-your-own-test)
in the official CKEditor documentation.

### This PR contains

- [x] Unit tests
- [x] Manual tests

## What changes did you make?

Added auto-selection for first autocomplete option on view open.

### CODE REVIEW

File: tests/plugins/autocomplete/manual/optionselection.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+@bender-tags: 4.10.0, feature, tp3558
```
Comment (mlewand) [inline tests/plugins/autocomplete/manual/optionselection.md]: It's a bug, not a feature.

File: tests/plugins/autocomplete/manual/optionselection.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+2. Type `@`
+
+## Expected
+
+First autocomplete option is selected.
+
+## Unexpected
+
+No option is selected.
+
```
Comment (mlewand) [inline tests/plugins/autocomplete/manual/optionselection.md]: There are two empty line at the end (even though GH shows only one), let's reduce it just to one.

File: tests/plugins/autocomplete/autocomplete.js [hunk 3]
```diff
@@ -89,7 +77,6 @@
 
 			editor.editable().fire( 'keyup', new CKEDITOR.dom.event( {} ) );
 
-			editable.fire( 'keydown', new CKEDITOR.dom.event( { keyCode: 40 } ) );
```
Comment (mlewand) [inline tests/plugins/autocomplete/autocomplete.js]: This change is not related to this issue 🤔

File: tests/plugins/autocomplete/autocomplete.js [hunk 4]
```diff
@@ -65,18 +59,12 @@
...
 			assertViewOpened( ac, true );
-			assert.isTrue( ac.view.getItemById( 2 ).hasClass( 'cke_autocomplete_selected' ) );
-
-			editable.fire( 'keydown', new CKEDITOR.dom.event( { keyCode: 38 } ) );
-
-			assertViewOpened( ac, true );
-
-			assert.isFalse( ac.view.getItemById( 2 ).hasClass( 'cke_autocomplete_selected' ) );
-			assert.isTrue( ac.view.getItemById( 1 ).hasClass( 'cke_autocomplete_selected' ) );
+			assert.isTrue( ac.view.getItemById( 3 ).hasClass( 'cke_autocomplete_selected' ) );
```
Comment (mlewand) [inline tests/plugins/autocomplete/autocomplete.js]: Could we check another subsequent key up press? Here the selection cycled to the end of list (which is important case to test), we're missing the "regular" go up case.

File: tests/plugins/autocomplete/autocomplete.js [hunk 5]
```diff
@@ -39,18 +39,12 @@
 
 			editor.editable().fire( 'keyup', new CKEDITOR.dom.event( {} ) );
 
-			assertViewOpened( ac, true );
-
-			editable.fire( 'keydown', new CKEDITOR.dom.event( { keyCode: 40 } ) );
-
 			assertViewOpened( ac, true );
 			assert.isTrue( ac.view.getItemById( 1 ).hasClass( 'cke_autocomplete_selected' ) );
```
Comment (mlewand) [inline tests/plugins/autocomplete/autocomplete.js]: You're asserting that the first item is focused in "test arrow down selects next item". Instead the bug that you have fixed should get a dedicated TC, like "test autocomplete starts with the first item selected".

Same issue happens in the "test arrow up selects previous item" TC, you're testing whether the fist item was focused there too.

File: tests/plugins/autocomplete/autocomplete.js (hunk 3 above)
Comment (jacekbogdanski) [reply tests/plugins/autocomplete/autocomplete.js]: After this bug fix I just simplified above test a little. There is no need to pass arrow down keyCode because an option is selected already.

Comment (mlewand) [reply tests/plugins/autocomplete/autocomplete.js]: Ah, OK, in that case it makes perfect sense - thanks for clarification!

File: tests/plugins/autocomplete/manual/__template__.html [hunk 6]
```diff
@@ -30,6 +26,10 @@ <h1>Lorem ipsum dolor sit amet, consectetur.</h1>
 		}
 	} );
 
+	// The body of the root html element has additional margin-left to display bender manual-sidebar which has fixed position.
+	// Our autcomplete view lives inside root html so with its margin it has invalid position relative to the editor.
+	// We have to change position of our editor relative to the root html by setting position:static ( see style tag ) and move it to the center of the page
+	// so it won't be hidden below sidebar.
```
Comment (mlewand) [inline tests/plugins/autocomplete/manual/__template__.html]: > autcomplete view lives inside root html

Technically it sits in `body` which is not a HTML root element.

Comment (mlewand) [inline tests/plugins/autocomplete/manual/__template__.html]: Still this workaround is way over-complicated. Am I correct to say that simply setting:
```html
<style>
	body {
		margin-left: 0px;
		padding-left: 350px;
	}
</style>
```

Solves the problem? Make sure to still leave a short comment for the reason of the workaround.

File: tests/plugins/autocomplete/autocomplete.js [hunk 7]
```diff
@@ -40,18 +54,22 @@
 			editor.editable().fire( 'keyup', new CKEDITOR.dom.event( {} ) );
 
 			assertViewOpened( ac, true );
+			assert.isTrue( ac.view.getItemById( 1 ).hasClass( 'cke_autocomplete_selected' ) );
```
Comment (mlewand) [inline tests/plugins/autocomplete/autocomplete.js]: Again, you're asserting here same thing as in "test autocomplete starts with the first item selected" TC. This is not needed.

File: tests/plugins/autocomplete/autocomplete.js [hunk 8]
```diff
@@ -65,19 +83,28 @@
...
```
Comment (mlewand) [inline tests/plugins/autocomplete/autocomplete.js]: Again, you're asserting here same thing as in "test autocomplete starts with the first item selected" TC. This is not needed.
